src/lib/dereference.js

```diff
@@ -2,20 +2,25 @@ import walkObject from './walkObject';
 
 const SRC_ATTR_PATTERN = /src$/;
 
-export default function dereference(container, dataSources) {
+export default function dereference(container, dataSources, deleteKeys) {
```

Comment:
I would prefer this to be a config field so we can add arbitrary functionality later on without needed to mess with positional parameters
```javascript
dereference(container, dataSources, config = {})
```